cmake/onnxruntime_providers.cmake

```diff
@@ -85,9 +85,6 @@ if(NOT onnxruntime_DISABLE_CONTRIB_OPS)
 endif()
 
 if (onnxruntime_USE_FEATURIZERS)
-  if(NOT MSVC)
-    set_source_files_properties(${onnxruntime_cpu_featurizers_cc_srcs} PROPERTIES COMPILE_FLAGS "-Wno-unknown-warning")
```

Comment:
You may add the following line to: https://github.com/microsoft/onnxruntime/blob/master/cmake/CMakeLists.txt#L502
```
  check_cxx_compiler_flag(-Wno-unknown-warning HAS_NO_UNKNOWN_WARNING)
```

Then at here you can use:
```
 if(HAS_NO_UNKNOWN_WARNING)	
    set_source_files_properties(${onnxruntime_cpu_featurizers_cc_srcs} PROPERTIES COMPILE_FLAGS "-Wno-unknown-warning")
```